labgrid/driver/usbloader.py

```diff
@@ -82,3 +83,54 @@ def load(self, filename=None):
             self.loader.command_prefix +
             [self.tool, "-p", str(self.loader.path), "-c", mf.get_remote_path()]
         )
+
+
+@target_factory.reg_driver
+@attr.s(cmp=False)
+class RKUSBDriver(Driver, BootstrapProtocol):
+    bindings = {
+        "loader": {RKUSBLoader, NetworkRKUSBLoader},
+    }
+
+    image = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+        # FIXME make sure we always have an environment or config
+        if self.target.env:
+            self.tool = self.target.env.config.get_tool('rk-usb-loader') or 'rk-usb-loader'
+        else:
+            self.tool = 'rk-usb-loader'
+
+    def on_activate(self):
+        pass
+
+    def on_deactivate(self):
+        pass
+
+    @Driver.check_active
+    @step(args=['filename'])
+    def load(self, filename=None):
+        # FIXME make sure we always have an environment or config
+        loaders = self.target.env.config.get_tool('loader') or 'loader'
```

Comment:
Loader seems to be a very generic name. Is this an additional image which needs to be loaded before the actual bootloader? If so, it should probably be set via an attribute on the driver (like the image itself).